### ISSUE TITLE
Dynamic SSL for HTTP listner transport is not working

### ISSUE THREAD

**Issue by madushadhanushka** [wso2/micro-integrator#2034]
**Description:**
Setting up a dynamic SSL profile does not work with MI for the HTTP listener. The same configurations does work for HTTP  sender.

**Steps to reproduce:**
Create two MI server, server 1 as the gateway and server 1 as the backend.
Add the following proxy configuration for server 1.
```
<proxy xmlns="http://ws.apache.org/ns/synapse"
       name="sender" transports="https,http" statistics="disable" trace="disable" startOnLoad="true"> <target>
    <inSequence>
        <log level="full"/>
    <call>
            <endpoint>
               <http method="get" uri-template="https://localhost:8254/services/listener"/>
            </endpoint>
         </call>
         <respond/>
    </inSequence>
    <outSequence>
    </outSequence>
</target>
    <description/>
</proxy>
```
Add the following proxy configuration for server 2.
```
<proxy xmlns="http://ws.apache.org/ns/synapse"
       name="listener" transports="https,http" statistics="disable" trace="disable" startOnLoad="true"> <target>
    <inSequence>
        <log level="full"/>
            <payloadFactory media-type="json">
                <format>
                    {
                    "hello": "world"
                    }
                </format>
                <args>
                </args>
            </payloadFactory>
            <respond/>
    </inSequence>
    <outSequence>
    </outSequence>
</target>
    <description/>
</proxy>
```
Add the following configuration on server 1 to setup HTTP sender with HTTPS protocol.
```
[transport.http]
sender.secured_enable = true                # inferred default true
sender.keystore.location ="repository/resources/security/selfsigned.jks" # inferred
sender.keystore.type = "JKS" # inferred
sender.keystore.password = "wso2carbon" # inferred
sender.keystore.key_password = "wso2carbon" # inferred
sender.truststore.location = "repository/resources/security/selfsigned.jks" # inferred
sender.truststore.type = "JKS" # inferred
sender.truststore.password = "wso2carbon" # inferred
```
Add the following configuration on server 2 to setup an HTTP listener with HTTP with an offset of 11.
```
[server]
offset  = 11
[transport.http]
listener.ssl_profile.file_path = "conf/sslprofiles/listenerprofiles.xml"
listener.ssl_profile.read_interval = 600000
```

Add the following listener configs into the `<EI_HOME>/conf/sslprofiles/listenerprofiles.xml` file.
```
<parameter name="SSLProfiles">
<profile>
        <bindAddress>192.168.43.255</bindAddress>
        <KeyStore>
            <Location>repository/resources/security/selfsigned.jks</Location>
            <Type>JKS</Type>
            <Password>wso2carbon2</Password>
            <KeyPassword>wso2carbon</KeyPassword>
            </KeyStore>
        <TrustStore>              
            <Location>repository/resources/security/selfsigned.jks</Location>
            <Type>JKS</Type>
            <Password>wso2carbon</Password>
        </TrustStore>
        <SSLVerifyClient>require</SSLVerifyClient>
    </profile>
</parameter>
```
Start the server and send a request to the server 1 proxy service. This will show SSL certificate not matching error.
**Note:**
The same issue tested for the sender profile as follows and those configs are working fine.
Set server 1 dynamic SSL profile as follows with TOML config.
```
[transport.http]
sender.ssl_profile.file_path = "conf/sslprofiles/senderprofiles.xml"
sender.ssl_profile.read_interval = 600000
```
Set server 2 HTTP listener SSL config as follows
```
[transport.http]
listener.keystore.location ="repository/resources/security/selfsigned.jks" # inferred
listener.keystore.type = "JKS" # inferred
listener.keystore.password = "wso2carbon" # inferred
listener.keystore.key_password = "wso2carbon" # inferred
```
Server 1 <MI_HOME>conf/sslprofiles/senderprofiles.xml file as follows.
```
<parameter name="customSSLProfiles">
    <profile>
        <servers>localhost:8254,192.168.43.255:8254</servers>
        <KeyStore>
            <Location>repository/resources/security/selfsigned.jks</Location>
            <Type>JKS</Type>
            <Password>wso2carbon</Password>
            <KeyPassword>wso2carbon</KeyPassword>
        </KeyStore>
        <TrustStore>          
            <Location>repository/resources/security/selfsigned.jks</Location>
            <Type>JKS</Type>
            <Password>wso2carbon</Password>
        </TrustStore>
    </profile>
</parameter>
```
selfsigned.jks self sign certificate can be generated with the following command.
```keytool -genkey -keyalg RSA -alias tomcat -keystore selfsigned.jks -validity 365 -keysize 2048 -ext SAN="dns:test.com,dns:foo.com,ip:192.168.43.255,ip:127.0.0.1,dns:localhost"```

Related issues https://github.com/wso2/docs-ei/issues/241

**Comment by madushadhanushka** [comment]
Doc removals
https://github.com/wso2/docs-ei/pull/2675
https://github.com/wso2/docs-ei/pull/2674
https://github.com/wso2/docs-ei/pull/2676

**Comment by sajinieKavindya** [comment]
In order to make the Dynamic SSL for HTTP listener transport works, we need to configure the "org.apache.synapse.transport.passthru.PassThroughHttpMultiSSLListener" over default transport receiver "org.apache.synapse.transport.passthru.PassThroughHttpSSLListener" in the axis2.xml. This can be done by configuring a custom_transport.listener in the deployment.toml after disabling the default transport receiver there.

Also, It is recommended to configure the hostname as the server when configuring custom SSL profiles. If an IP address is required to be configured as the server, the IP address needs to be mapped to a hostname in the Host file, and the hostname can be provided as the server.

Created a doc issue [1] to provide the correct steps to enable Dynamic SSL profiles. 
 [1] - https://github.com/wso2/docs-ei/issues/2757

**Comment by cabakiter** [comment]
Hi there, I see you're quite experts! I'm struggling with first WSO MI tests. What I'm trying to achieve is to build an API under WSO Integration Studio which calls an endpoint requiring SSL verification. When querying the endpoint, namely https://37.186.199.242:6443/web-app/ by chrome, it asks for a certificate (say cert.p12) which I received (together with its password) and installed. I can invoke the same endpoint i.e. by python with the following: 
response = requests.get(request_URL, data=data, headers=self.headers, cert=(certfile,keyfile), verify=False, proxies=self.proxies) where certfile and keyfile were generated from cert.p12 file via openssl command. I tried to implement what suggested here: https://ei.docs.wso2.com/en/latest/micro-integrator/setup/transport_configurations/multi-https-transport/ , but when calling the API via POSTMAN over WSO I got in consolle error "pkix path building failed". Is there a step by step procedure to set up WSO Micro Integrator to correctly connect to the endpoint? Thanks in advance
